app/2.0/docs/tools/documentation.md

```diff
@@ -4,10 +4,302 @@ title: Document your elements
+     * In typical use, just slap some `<awesome-sauce>` at the top of your body:
+     * <body>
+     *   <awesome-sauce></awesome-sauce>
+     * Wham! It's all awesome now!
+     * @customElement
+     * @polymer
+     * @demo https://path/to/awesomeness/demo/
+     * 
+     */
+    class AwesomeSauce extends Polymer.Element { 
+      ... 
+    }
+  ```
+  
+The first tag encountered in the comment block marks the end of the element
+summary. **Any line starting with an at-sign (@) is interpreted as a tag.** Any
+remaining non-tag comments in the comment block are ignored.
+
+### Properties {#properties}
+
+<p class="tldr">
+  Document all public properties. Docs should start with a one line
+  summary. Make sure that the property's type is annotated.
+</p>
+
+Denote your property documentation with a **JavaScript comment** preceding the property declaration. The simplest property documentation can be a single
+line:
+
+```js
+/** Whether this element is currently awesome. */
+isAwesome: Boolean,
+```
+
+If the property doesn't specify a type, or that type is not primitive,
+be sure to [annotate](#type-annotation) the type properly:
+
+```js
+/**
+ * Metadata describing what has been made awesome on the page.
+ *
+ * @type {{elements: Array<HTMLElement>, level: number}}
+ */
+sauce: Object,
+```
+
+Private properties should be prefixed with an underscore (`_`):
+
+```js
+/** An awesome message */
+_message: String,
+```
+
+### Methods {#methods}
+ 
+Follow the [property guidelines](#properties). Additionally, make sure
+the types for all params and return values are documented.
+{ .tldr }
+
+For example:
+
+```js
+/**
+ * Applies awesomeness to `element`.
+ *
+ * @param {HTMLElement} element The element to be made awesome.
+ * @param {number} level The numeric level of awesomeness. A value
+ *     between `1` and `11`.
+ * @param {Array<HTMLElements>=} refs Optional referenced elements
+ *     that become awesome by proxy.
+ * @return {number} The cumulative level of awesomeness.
+ */
+makeAwesome: function makeAwesome(element, level, refs) {
+```
+
+### Events {#events}
+
+<p class="tldr">
+  Events must be annotated explicitly with an <code>@event</code> tag.
+</p>
+
+Event properties are documented with the `@param` tag, just like method parameters.
+
+For example:
+
+```js
+/**
+ * Fired when `element` changes its awesomeness level.
+ *
+ * @event awesome-change
+ * @param {number} newAwesome New level of awesomeness.
+ */
+```
+
+### Behaviors {#behaviors}
+
+<p class="tldr">Like an element, but add <code>@polymerBehavior</code>.</p>
+
+Include a behavior summary, just like an element summary, but ending with a
+`@polymerBehavior` tag. The behavior name can be specified explicitly if the
+doc parser can't infer it correctly.
+
+    @polymerBehavior MyOddBehavior
+
+Document methods, properties, etc. just like an element.
+
+For example:
+
+```js
+/**
+ * Behavior that highlights stuff.
+ *
+ * @polymerBehavior
+ */
+
```

Comment:
Since https://github.com/Polymer/polymer-analyzer/pull/696, it's required that the JSDoc comment blocks are actually touching the symbol they are associated with, so there shouldn't be a newline here.

(BTW I'm wondering if that analyzer PR was a little overkill -- it does help eliminate false associations though.)